api/python/cellxgene_census/src/cellxgene_census/_util.py

```diff
@@ -18,3 +20,9 @@ def _uri_join(base: str, url: str) -> str:
         p_url.fragment,
     ]
     return urllib.parse.urlunparse(parts)
+
+
+def _extract_census_version(census: soma.Collection) -> str:
```

Comment:
this doesn't handle errors very well.  Previously, the API worked fine for Census builds that were local (e.g., opened with the `uri=` param), and I suspect we _really_ don't want to break that as it rules out people taking local copies.

I'd give some thought to how you know if the "version" returned is nonsense?  (e.g., the SOMA Census objects are copied into /tmp/foobar).  At least you inspect it and see if it looks like a YYYY-MM-DD, and throw an error if not.

You could also sniff the `census.uri` prefix, and see if it is in an expected location (rather than file:///tmp/foobar/)

I think raising is perfectly fine here, as it just means that some edge cases can't infer the version from the URI, and therefore are unsupported (with a nice clean error saying so)

Reply:
I also think raising is fine. This is an edge case because right now this is only called in `get_anndata` when adding the embeddings and nowhere else. In this case, we can't guarantee that the local copy of the Census matches the embeddings, therefore raising is the only option.